### PR TITLE
Use name from ORCID in author/contributor form

### LINKED ISSUE #136: Implement ORCID support
Review the implementation of ORCID in [H2](https://github.com/search?q=repo%3Asul-dlss%2Fhappy-heron%20orcid&type=code). ORCID support is a requirement for our first beta user.

# TO-DO

- [x] ORCID radio button should be the default 
- [x] "https://orcid.org" should already be in the field, to help them enter the entire thing correctly (overwrite when being auto-populated).
- [x] Once an ORCID is entered, the lookup should occur and populate the Name fields for first name and last name.

### PR DESCRIPTION
**Pull request by mjgiarlo**
Fixes #136 

- **Add resolver to pull names from ORCID API**
- **Enter user names in author/contributor form as registered in orcid**

Peek 2024-12-16 13-52

### CODE REVIEW

**Review by justinlittman — APPROVED** [review]
Approve pending consideration of comments.

File: app/services/orcid_resolver.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  end
+
+  def initialize(orcid_id:)
+    @orcid_id = orcid_id
+  end
+
+  def call
+    return orcid_result if orcid_result.failure?
+
+    response = Faraday.new.get(url, {}, headers)
```
[the remark this answers is not in the document]
Comment (mjgiarlo) [reply app/services/orcid_resolver.rb:19]: I doubt it. (Copypasta from H2.)

File: app/javascript/controllers/contributors_controller.js [hunk 2]
```diff
@@ -39,6 +42,38 @@ export default class extends Controller {
...
+        if (error.message === '404') {
+          this.showOrcidError('ORCID iD not found')
+        } else {
+          console.dir(error)
+          this.showOrcidError('Error validating ORCID iD. Please clear and try again.')
+        }
+      })
+  }
+
+  showOrcidError (error) {
```
[the remark this answers is not in the document]
Comment (mjgiarlo) [reply app/javascript/controllers/contributors_controller.js]: I think so? Is this far off? @justinlittman 

Peek 2024-12-16 14-05

File: app/services/orcid_resolver.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (mjgiarlo) [reply app/services/orcid_resolver.rb]: Removed

File: app/javascript/controllers/contributors_controller.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (justinlittman) [reply app/javascript/controllers/contributors_controller.js]: This interaction doesn't seem right:

https://github.com/user-attachments/assets/65838ede-4de9-4919-8a15-7ae13e6a390e

[the remark this answers is not in the document]
Comment (mjgiarlo) [reply app/javascript/controllers/contributors_controller.js]: @justinlittman Which part(s) don't seem right?

[the remark this answers is not in the document]
Comment (justinlittman) [reply app/javascript/controllers/contributors_controller.js]: The validation is inconsistent. Sometimes it is validated locally. Sometimes it is validated on the server.

I also don't like how the right button changes. I totally missed that at first and couldn't edit the orcid.

[the remark this answers is not in the document]
Comment (mjgiarlo) [reply app/javascript/controllers/contributors_controller.js]: @justinlittman The button should not have changed. Will fix that bug.

I'm going to try ditching the client-side validation too.

[the remark this answers is not in the document]
Comment (mjgiarlo) [reply app/javascript/controllers/contributors_controller.js]: This is better now, thanks to your keen eye. I added a couple bits:

* When form renders, make sure the text field is read-only if the user has already clicked the "use my orcid" button.
* Improve the logic in the form to determine when the render the use & reset buttons.
* Remove client-side validation

**Pull request merged**